app/controllers/activities_controller.rb

```diff
@@ -7,10 +7,11 @@ class ActivitiesController < AuthenticatedController
   def index
     @page = params[:page].present? ? params[:page].to_i : 1
 
-    activities = current_project.activities
-                                .includes(:trackable)
-                                .order(created_at: :desc)
-                                .page(@page)
+    activities = current_project.activities.includes(:trackable)
+    activities = activities.filter_by_user_id(params[:user]) if params[:user].present?
+    activities = activities.filter_by_type(params[:type]) if params[:type].present?
+    activities = activities.filter_by_date(period_start, period_end) if params[:period_start].present?
```

Comment:
What about moving these to a helper method so that the index action is less cluttered? 
Ex. `activities = filter_activities`